### PR TITLE
AI Tutor - Recommendation&Chat - UI Implementierung

### PR DESCRIPTION
**Pull request by JonathanWolp**
# Description of changes
Ticket:
https://gamify-it.gropius.dev/projects/ebc15ee5-70fe-49b6-9c3f-6e7dd23a1304/issues/0203ab75-ef55-4fd3-a2e2-bbd1b71228d3

Please provide a short description of the change or if the change is sufficiently described in the GitHub issue, provide a link to the issue.

## Beschreibung

Diese Änderung implementiert die Grundfunktionalität des AI-Tutors für die Chat-Komponente sowie die Recommendation-Blase im Frontend.  

Umgesetzte Features:

- **Chat-Komponente**
  - UI für den Nutzer-Chat inkl. Eingabefeld, Sende-Button und Anzeige vergangener Nachrichten
  - Grundlegende State-Verwaltung für Chatverlauf (Dummy-/Platzhalterdaten)
  - Tutor per Drag and Drop bewegen (an drei verschiedene Stellen zu bewegen an der rechten Seite)
  - API Schnittstelle angelegt

- **Recommendation-Blase**
  - UI-Element, das Empfehlungen bzw. weiterführende Hinweise im Chatverlauf anzeigt
  - Platzhalter für spätere dynamische Inhalte
  - Einbindung in den Chat-Kontext
  - Aufruf aus dem Backend per API möglich

Hinweise:

- Die Anbindung an die API erfolgt in einem späteren Schritt/Merge Request.
- Alle API-Calls sind aktuell gemockt bzw. noch nicht implementiert.
- Das Ziel dieses MRs ist die Bereitstellung der Benutzeroberfläche und der Kerninteraktionen, damit Frontend-Tests und UI-Reviews möglich sind.

## How has this been tested?

Please describe the test strategy you followed.

- [ ] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

In case of manual test, please document the test well including a set of user instructions and prerequisites. Each including an action, it's result, and where appropriate a screenshot.

## Checklist before requesting a review

- [x] My code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfilles all acceptance criteria
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have made corresponding changes to the documentation
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/MEITREX/wiki/tree/main/adr)
- [x ] I have updated the changes in the ticket description

## Checklist for reviewer

- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/MEITREX/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: components/tutor/TutorChat.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    // Sendet POST an /api/llm mit { prompt }, erwartet Antwort { answer }.
+    // /api/llm muss den Prompt an das LLM (z.B. Ollama) weiterleiten und { answer } zurückgeben.
+    try {
+      // Anfrage an den Bot (API ggf. anpassen!)
+      const response = await fetch("/api/llm", {
+        // z.B. const response = await fetch('http://localhost:8000/api/classify_and_preprocess', {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({ prompt }),
+      });
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx:62]: Hast du vollkommen recht!

File: components/tutor/TutorWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+let widgetApi: TutorWidgetApi | null = null;
+
+export function showTutorRecommendation(rec: Recommendation) {
+  if (widgetApi) widgetApi.showRecommendation(rec);
+}
+
+export function clearTutorRecommendations() {
+  if (widgetApi) widgetApi.clearRecommendations();
+}
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorWidget.tsx:49]: wurde aufgeräumt

File: components/Navbar.tsx [hunk 3]
```diff
@@ -424,13 +424,14 @@ function UserInfo({ _isTutor }: { _isTutor: NavbarIsTutor$key }) {
               <IconButton
                 edge="end"
                 aria-label="logout"
-                onClick={() =>
+                onClick={() => {
+                  window.localStorage.removeItem("meitrex-welcome-shown");
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/Navbar.tsx:428]: Wurde rausgenommene, dass es immer angezeigt wird

File: components/tutor/TutorWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+          aria-label={open ? "Tutor einklappen" : "Tutor öffnen"}
+        >
+          <TutorAvatar />
+        </button>
+      </div>
+      {/* Chat-Blase, nach links aufklappend */}
+      {open && (
+        <div
+          style={{
+            background: "#fff",
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorWidget.tsx:339]: auskommentiert, falls doch mit dark theme

File: components/tutor/TutorWidget.tsx [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+            borderRadius: "18px 0 18px 18px",
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorWidget.tsx]: wurde übernommen

File: components/tutor/TutorChat.tsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+            resize: "none",
+            minHeight: 36,
+            maxHeight: 90,
+            overflowY: "auto",
+            fontFamily: "inherit",
+            fontSize: 15,
+          }}
+          disabled={loading}
+          rows={1}
+        />
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx]: wurde durch material UI ersetzt

File: components/tutor/TutorChat.tsx [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+// Message-Typ für den Chatverlauf
+type Message = {
+  sender: "user" | "bot";
+  text: string;
+};
+
+const TutorChat: React.FC = () => {
+  const [input, setInput] = useState("");
+  const [chatHistory, setChatHistory] = useState<Message[]>([]);
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx:19]: Würde ich in einem nächsten Schritt anpassen und erstmal so belassen

File: components/tutor/TutorChat.tsx [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+          return updated;
+        });
+        setLoading(false);
+        sendTimestamp.current = null; // <- wichtig!
+      }, waitMore);
+    }
+  };
+
+  // Optional: Mit Enter senden, mit Shift+Enter Zeilenumbruch
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx]: Wurde übernommen

File: components/tutor/TutorChat.tsx [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+
+  // Automatisches Anpassen der textarea-Höhe (Auto-Resize)
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInput(e.target.value);
+
+    if (textareaRef.current) {
+      textareaRef.current.style.height = "auto";
+      textareaRef.current.style.height =
+        textareaRef.current.scrollHeight + "px";
+    }
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx]: wurde übernommen

File: components/tutor/TutorChat.tsx [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+          </div>
+        )}
+        {chatHistory.map((msg, idx) => (
+          <div
+            key={idx}
+            style={{
+              display: "flex",
+              justifyContent: msg.sender === "user" ? "flex-start" : "flex-end",
+              margin: "6px 0",
+              width: "100%",
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx]: ja stimmt, wurde rausgemacht

File: components/tutor/TutorChat.tsx [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+            }}
+          >
+            <span
+              style={{
+                display: "inline-block",
+                background: msg.sender === "user" ? "#e3f2fd" : "#f0f4c3",
+                borderRadius: 12,
+                padding: "6px 10px",
+                maxWidth: 200,
+                wordBreak: "break-word",
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx]: angepasst

File: components/tutor/TutorChat.tsx [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+            updated.push({
+              sender: "bot",
+              text: data.answer ?? "Fehler oder keine Antwort erhalten.",
+            });
+          }
+          return updated;
+        });
+        setLoading(false);
+        sendTimestamp.current = null; // <- wichtig!
+      }, waitMore);
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorChat.tsx]: wurde angepasst

File: public/DinoTutor.png
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply public/DinoTutor.png:1]: wurde in den chat eingefügt

File: components/tutor/TutorWidget.tsx [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+  /* Recommendations API Logic
+import { showTutorRecommendation, clearTutorRecommendations } from "@/components/tutor/TutorWidget";
+
+// Zeige eine Empfehlung:
+showTutorRecommendation({ id: "empfehlung-1", text: "Teste die neue Quiz-Funktion!" });
+
+// Empfehlungen wieder schließen:
+clearTutorRecommendations();
+
+*/
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorWidget.tsx]: ist raus

File: components/tutor/TutorWidget.tsx [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+
+  // Welcome nur beim Login und nur, wenn er noch nicht angezeigt wurde
+  useEffect(() => {
+    if (!isAuthenticated) return;
+    const alreadyShown = window.localStorage.getItem(WELCOME_KEY);
+    if (!alreadyShown) {
+      setShowWelcome(true);
+      window.localStorage.setItem(WELCOME_KEY, "true");
+    }
+  }, [isAuthenticated]);
```
[the remark this answers is not in the document]
Comment (JonathanWolp) [reply components/tutor/TutorWidget.tsx]: ist raus

**Comment by JonathanWolp** [comment]
Doku:
https://meitrex.readthedocs.io/en/latest/dev-manuals/frontend/ai_tutor.html
